packaging/os/yum.py

```diff
@@ -763,26 +792,53 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
                 res['msg'] += "The following packages have pending transactions: %s" % ", ".join(conflicts)
                 module.fail_json(**res)
 
-            pkg = spec
-        if not cmd:
-            cmd = yum_basecmd + [basecmd, pkg]
+    # check_mode output
+    if module.check_mode:
+        if update_all:
+            will_update = updates.keys()
+        else:
+            will_update = [u for u in updates.keys() if u in pkgs['update']]
 
-        if module.check_mode:
-            return module.exit_json(changed=True)
+        for w in will_update:
+            msg = '{0} will be updated with {0}-{1}.{2} from {3}'.format(w, updates[w]['version'],  updates[w]['dist'], updates[w]['repo'])
```

Comment:
This module needs to be python-2.4 compatible so we can't use format.  Something like this should work: 
```
msg = '%s will be updated with %s-%s.%s from %s'.format(w, w, updates[w]['version'],  updates[w]['dist'], updates[w]['repo'])
```